syntax = "proto3";

package mqtt;

// MQTT Bronze Record Schema
// Matches the Databricks Unity Catalog table schema for MQTT data
message MQTTBronzeRecord {
    // Timestamps in microseconds (int64 for large values)
    int64 event_time = 1;
    int64 ingest_time = 2;

    // Source information
    string source_name = 3;

    // MQTT specific fields
    string topic = 4;

    // Sensor identification
    string industry = 5;
    string sensor_name = 6;

    // Value information
    double value = 7;
    string value_string = 8;
    string unit = 9;
    string sensor_type = 10;

    // MQTT metadata
    int64 timestamp_ms = 11;
    int32 qos = 12;
    bool retain = 13;
}
